src/generic/NetworkManager.cpp

```diff
@@ -36,6 +42,15 @@ void NetworkManager::send(RadioPacket &packet) {
 }
 
 void NetworkManager::AddToRecentIdsCache(uint16_t id) {
-  recent_ids_cache_[recent_ids_cache_index_++] = id;
-  recent_ids_cache_index_ %= kRecentIdsCacheSize;
+  uint16_t sum = 0;
+  for (uint8_t n = 0; n < kRecentIdsCacheSize; ++n) {
+    sum += recent_ids_num_seen_[n];
+  }
+  if (sum > kPacketCopiesThreshold) {
+    resume_rebroadcast_at = millis() + kRebroadcastPauseMillis;
+  }
+
+  recent_ids_cache_[recent_ids_cache_index_] = id;
+  recent_ids_num_seen_[recent_ids_cache_index_] = 0;
+  recent_ids_cache_index_ = (recent_ids_cache_index_ + 1) % kRecentIdsCacheSize;
```

Comment:
Nit: Not from this PR but this name is odd, maybe `recent_ids_cache_cursor_` or something.